File: SCC0600/exercíciosPy/ex27.py
n = int(input())

casas = 0
soma = 0

while 10**casas <= n:
  casas = casas + 1

rcounter = casas
new = n

for _ in range(casas):
  digito = new // 10**(rcounter - 1)
  new = new - digito * 10**(rcounter - 1)
  soma = soma + digito
  rcounter = rcounter - 1

print(soma)

File: SCC0600/exerciciosC/ex27.c
#include <stdio.h>
#include <math.h>

int modulo (int value) {
    if (value < 0) {
        return -value;
    }
  return value;  
}

int somadigitos (long n) {
    int casas = 0;
    while (pow(10, casas) <= n) {
        casas++;
    }

    int soma = 0;
    long long newnum = n;
    for (int i = casas - 1; i >= 0; i--) {
        long double pot = pow(10, i);
        int digito = newnum / pot;
        newnum -= digito * pot;
        soma += digito;
    }

    return soma;
}

int main (void) {
    long long a;
    scanf("%ld", &a);
    printf("%d\n", somadigitos(modulo(a)));
}